### PR TITLE
Documentation update: How to use the clang toolchain

### PR DESCRIPTION
**Pull request by sigmaaa**
added documentation about how to use the clang toolchain

### CODE REVIEW

**Review by kolipakakondal — CHANGES_REQUESTED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -363,6 +364,70 @@ We now need to tell CDT which toolchain to use when building the project. This w
...
+  is not a full path and was not found in the PATH.
+```
+To fix this you can use clang instead of clang++. Edit a file `esp-idf\tools\cmake\toolchain-clang-esp32.cmake`:
+```diff
+set(CMAKE_C_COMPILER clang)
+- set(CMAKE_CXX_COMPILER clang++)
++ set(CMAKE_CXX_COMPILER clang)
+set(CMAKE_ASM_COMPILER clang)
+
+```
```
Comment (kolipakakondal) [inline README.md:430]: For documenting workarounds, could you please use [this](https://github.com/espressif/idf-eclipse-plugin/blob/master/WORKAROUNDS.md) and link it here

File: README.md [hunk 2]
```diff
@@ -363,6 +364,70 @@ We now need to tell CDT which toolchain to use when building the project. This w
 
 ![](docs/images/7_cmake_toolchain.png)
 
+<a name="SelectDifferentToolchain"></a>
+# Selecting Clang Toolchain
+In ESP-IDF Eclipse Plugin v2.7.0 you can build your project with Clang Toolchain.
```
Comment (kolipakakondal) [inline README.md:369]: With ESP-IDF Eclipse Plugin v2.7.0 and higher you can build your project with Clang Toolchain

File: README.md (hunk 1 above)
Comment (sigmaaa) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (sigmaaa) [reply README.md]: fixed. Thanks

**Review by kolipakakondal — APPROVED** [review]
Thanks @sigmaaa for the PR. LGTM

**Pull request merged**